target/linux/lantiq/dts/RS353.dts

```diff
@@ -0,0 +1,318 @@
+/dts-v1/;
+
+/include/ "vr9.dtsi"
+
+/ {
+	model = "RS353 - Bintec Elmeg RS353";
+
+	chosen {
+		bootargs = "console=ttyLTQ0,115200 init=/etc/preinit";
+		leds {
+                        running = &status;
+                        boot = &status;
+			wifi = &wlan;
+			usb2 = &usb; /* Front port is mapped to device 2 */
+			dsl = &dsl;
+                };
+
+	};
+
+	memory@0 {
+		reg = <0x0 0x8000000>;
+	};
+
+	sram@1F000000 {
+		cgu0: cgu@103000 {
+                        compatible = "lantiq,cgu-xway";
+                        reg = <0x103000 0x1000>;
+                        #clock-cells = <1>;
+			lantiq,phy-clk-src = <0x2>;
+		};
+	};
+
+	fpi@10000000 {
+		localbus@0 {
+			nor-boot@0 {
+				compatible = "lantiq,nor";
+				bank-width = <2>;
+				reg = <0 0x0 0x2000000>;
+				#address-cells = <1>;
+				#size-cells = <1>;
+
+				partitions {
+					compatible = "fixed-partitions";
+					#address-cells = <1>;
+					#size-cells = <1>;
+
+					boardconfig: partition@0 {
```

Comment:
OK, I found something about the uart issue: 

In the kernel source, in file build_dir/target-mips_34kc+dsp_musl-1.1.15/linux-lantiq_xrx200/linux-4.4.14/arch/mips/lantiq/early_printk.c

I commented this:
```
if (c == '\n')
                ltq_w8('\r', LTQ_ASC_TBUF);
```
Now, I get the complete boot log without any issue:

```
OpenWrt kernel loader for MIPS based SoC
Copyright (C) 2011 Gabor Juhos <juhosg@openwrt.org>
Decompressing kernel... done!
Starting kernel at 80002000...

[    0.000000] Linux version 4.4.14 (qa@serveurQA) (gcc version 5.3.0 (OpenWrt GCC 5.3.0 50020) ) #11 SMP Tue Jan 10 21:23:29 UTC 2017
[    0.000000] SoC: xRX200 rev 1.2
[    0.000000] bootconsole [early0] enabled
[    0.000000] CPU0 revision is: 00019556 (MIPS 34Kc)
[    0.000000] MIPS: machine is RS353 - Bintec Elmeg RS353
[    0.000000] Determined physical RAM map:
[    0.000000]  memory: 08000000 @ 00000000 (usable)
[    0.000000] Initrd not found or empty - disabling initrd
[    0.000000] Zone ranges:
[    0.000000]   Normal   [mem 0x0000000000000000-0x0000000007ffffff]
[    0.000000] Movable zone start for each node
[    0.000000] Early memory node ranges
[    0.000000]   node   0: [mem 0x0000000000000000-0x0000000007ffffff]
[    0.000000] Initmem setup node 0 [mem 0x0000000000000000-0x0000000007ffffff]
[    0.000000] Detected 1 available secondary CPU(s)
[    0.000000] Primary instruction cache 32kB, VIPT, 4-way, linesize 32 bytes.
[    0.000000] Primary data cache 32kB, 4-way, VIPT, cache aliases, linesize 32 bytes
[    0.000000] PERCPU: Embedded 9 pages/cpu @8110c000 s8128 r8192 d20544 u36864
[    0.000000] Built 1 zonelists in Zone order, mobility grouping on.  Total pages: 32512
[    0.000000] Kernel command line: console=ttyLTQ0,115200 init=/etc/preinit
[    0.000000] PID hash table entries: 512 (order: -1, 2048 bytes)
[    0.000000] Dentry cache hash table entries: 16384 (order: 4, 65536 bytes)
[    0.000000] Inode-cache hash table entries: 8192 (order: 3, 32768 bytes)
[    0.000000] Writing ErrCtl register=0007edf0
[    0.000000] Readback ErrCtl register=0007edf0
[    0.000000] Memory: 122972K/131072K available (4019K kernel code, 168K rwdata, 1172K rodata, 1216K init, 222K bss, 8100K reserved, 0K cma-reserved)
[    0.000000] SLUB: HWalign=32, Order=0-3, MinObjects=0, CPUs=2, Nodes=1
[    0.000000] Hierarchical RCU implementation.
[    0.000000] NR_IRQS:256
[    0.000000] Setting up vectored interrupts
[    0.000000] Setting up IPI vectored interrupts
[    0.000000] CPU Clock: 500MHz
[    0.000000] clocksource: MIPS: mask: 0xffffffff max_cycles: 0xffffffff, max_idle_ns: 7645041786 ns
[    0.000011] sched_clock: 32 bits at 250MHz, resolution 4ns, wraps every 8589934590ns
[    0.007753] Calibrating delay loop... 332.54 BogoMIPS (lpj=665088)
[    0.042243] pid_max: default: 32768 minimum: 301
[    0.047007] Mount-cache hash table entries: 1024 (order: 0, 4096 bytes)
[    0.053468] Mountpoint-cache hash table entries: 1024 (order: 0, 4096 bytes)
[    0.064367] Primary instruction cache 32kB, VIPT, 4-way, linesize 32 bytes.
[    0.064384] Primary data cache 32kB, 4-way, VIPT, cache aliases, linesize 32 bytes
[    0.064636] CPU1 revision is: 00019556 (MIPS 34Kc)
[    0.123991] Synchronize counters for CPU 1: done.
[    0.128797] Brought up 2 CPUs
[    0.136416] clocksource: jiffies: mask: 0xffffffff max_cycles: 0xffffffff, max_idle_ns: 7645041785100000 ns
[    0.146182] pinctrl core: initialized pinctrl subsystem
[    0.152092] NET: Registered protocol family 16
[    0.162460] pinctrl-xway 1e100b10.pinmux: Init done
[    0.168497] dma-xway 1e104100.dma: Init done - hw rev: 7, ports: 7, channels: 28
[    0.281526] dcdc-xrx200 1f106a00.dcdc: Core Voltage : 1016 mV
[    0.445033] usbcore: registered new interface driver usbfs
[    0.450565] usbcore: registered new interface driver hub
[    0.455936] usbcore: registered new device driver usb
[    0.461634] PCI host bridge to bus 0000:00
[    0.465625] pci_bus 0000:00: root bus resource [mem 0x1c000000-0x1cffffff]
[    0.472422] pci_bus 0000:00: root bus resource [io  0x1d800000-0x1d8fffff]
[    0.479257] pci_bus 0000:00: root bus resource [??? 0x00000000 flags 0x0]
[    0.486010] pci_bus 0000:00: No busn resource found for root bus, will use [bus 00-ff]
[    0.493977] ifx_pcie_rc_class_early_fixup: fixed pcie host bridge to pci-pci bridge
[    0.513406] pci 0000:00:00.0: BAR 8: assigned [mem 0x1c000000-0x1c0fffff]
[    0.520081] pci 0000:00:00.0: BAR 9: assigned [mem 0x1c100000-0x1c1fffff pref]
[    0.527268] pci 0000:01:00.0: BAR 0: assigned [mem 0x1c000000-0x1c01ffff 64bit]
[    0.534539] pci 0000:01:00.0: BAR 6: assigned [mem 0x1c100000-0x1c10ffff pref]
[    0.541685] pci 0000:00:00.0: PCI bridge to [bus 01]
[    0.546586] pci 0000:00:00.0:   bridge window [mem 0x1c000000-0x1c0fffff]
[    0.553342] pci 0000:00:00.0:   bridge window [mem 0x1c100000-0x1c1fffff pref]
[    0.560516] ifx_pcie_bios_map_irq port 0 dev 0000:00:00.0 slot 0 pin 1 
[    0.567067] ifx_pcie_bios_map_irq dev 0000:00:00.0 irq 144 assigned
[    0.573289] ifx_pcie_bios_map_irq port 0 dev 0000:01:00.0 slot 0 pin 1 
[    0.579830] ifx_pcie_bios_map_irq dev 0000:01:00.0 irq 144 assigned
[    0.587849] clocksource: Switched to clocksource MIPS
[    0.595562] NET: Registered protocol family 2
[    0.600834] TCP established hash table entries: 1024 (order: 0, 4096 bytes)
[    0.607690] TCP bind hash table entries: 1024 (order: 1, 8192 bytes)
[    0.613990] TCP: Hash tables configured (established 1024 bind 1024)
[    0.620387] UDP hash table entries: 256 (order: 1, 8192 bytes)
[    0.626144] UDP-Lite hash table entries: 256 (order: 1, 8192 bytes)
[    0.632645] NET: Registered protocol family 1
[    0.637482] gptu: totally 6 16-bit timers/counters
[    0.642330] gptu: misc_register on minor 63
[    0.646490] gptu: succeeded to request irq 126
[    0.650879] gptu: succeeded to request irq 127
[    0.655285] gptu: succeeded to request irq 128
[    0.659691] gptu: succeeded to request irq 129
[    0.664097] gptu: succeeded to request irq 130
[    0.668505] gptu: succeeded to request irq 131
[    0.674467] phy-xrx200 gphy-xrx200: requesting lantiq/vr9_phy11g_a2x.bin
[    0.681791] phy-xrx200 gphy-xrx200: booting GPHY0 firmware at 7A00000
[    0.688133] phy-xrx200 gphy-xrx200: booting GPHY1 firmware at 7A00000
[    0.797690] futex hash table entries: 512 (order: 2, 16384 bytes)
[    0.822480] squashfs: version 4.0 (2009/01/31) Phillip Lougher
[    0.828231] jffs2: version 2.2 (NAND) (SUMMARY) (LZMA) (RTIME) (CMODE_PRIORITY) (c) 2001-2006 Red Hat, Inc.
[    0.840985] io scheduler noop registered
[    0.844914] io scheduler deadline registered (default)
[    0.851025] 1e100c00.serial: ttyLTQ0 at MMIO 0x1e100c00 (irq = 112, base_baud = 0) is a lantiq,asc
[    0.859901] console [ttyLTQ0] enabled
[    0.859901] console [ttyLTQ0] enabled
[    0.867164] bootconsole [early0] disabled
[    0.867164] bootconsole [early0] disabled
[    0.877006] lantiq nor flash device: 02000000 at 10000000
[    0.882728] ltq-nor 10000000.nor-boot: probing failed
[    0.995464] libphy: lantiq,xrx200-mdio: probed
[    1.068882] eth0: attached PHY [Lantiq XWAY VR9 GPHY 11G v1.4] (phy_addr=0:13, irq=-1)
[    1.140878] eth0: attached PHY [Lantiq XWAY VR9 GPHY 11G v1.4] (phy_addr=0:11, irq=-1)
[    1.212910] eth0: attached PHY [Lantiq XWAY PEF7071] (phy_addr=0:00, irq=-1)
[    1.284893] eth0: attached PHY [Lantiq XWAY PEF7071] (phy_addr=0:01, irq=-1)
[    1.356894] eth0: attached PHY [Lantiq XWAY PEF7071] (phy_addr=0:02, irq=-1)
[    1.365963] wdt 1f8803f0.watchdog: Init done
[    1.374910] NET: Registered protocol family 10
[    1.381108] NET: Registered protocol family 17
[    1.385634] bridge: automatic filtering via arp/ip/ip6tables has been deprecated. Update your scripts to load br_netfilter if you need this.
[    1.398188] 8021q: 802.1Q VLAN Support v1.8
[    1.408072] VFS: Cannot open root device "(null)" or unknown-block(0,0): error -6
[    1.414174] Please append a correct "root=" boot option; here are the available partitions:
[    1.422526] Kernel panic - not syncing: VFS: Unable to mount root fs on unknown-block(0,0)
[    1.432759] ---[ end Kernel panic - not syncing: VFS: Unable to mount root fs on unknown-block(0,0)
[   31.283861] random: nonblocking pool is initialized
```
Maybe an interrupt or clock issue ?